scripts/init.groovy.d/mesos-auth.groovy

```diff
@@ -0,0 +1,51 @@
+import com.cloudbees.plugins.credentials.*
+import com.cloudbees.plugins.credentials.domains.*
+import com.cloudbees.plugins.credentials.impl.*
+import hudson.tasks.*
+import jenkins.model.*
+import org.jenkinsci.plugins.mesos.MesosCloud
+
+def changePassword = { userName ->
+  def cloud = MesosCloud.get()
+  def credentialsId = cloud.getCredentialsId()
+  def credId = "mesos-${userName}"
+
+  if (credentialsId && credentialsId == credId) {
+    // do nothing if credential already exists
+    println "--> [mesos] credentials already selected"
+  } else {
+    // create a new credential with an expected ID
+    println "--> [mesos] creating new credentials"
+    String randomPwd = org.apache.commons.lang.RandomStringUtils.random(9, true, true)
+
+    mesosFrameworkCreds = new UsernamePasswordCredentialsImpl(
+      CredentialsScope.GLOBAL,
+      "mesos-${userName}",
+      "mesos authentication",
+      userName, randomPwd)
+    SystemCredentialsProvider.getInstance().getStore().addCredentials(Domain.global(), mesosFrameworkCreds)
+    cloud.setCredentialsId(mesosFrameworkCreds.getId())
+    Jenkins.getInstance().save()
+    cloud.restartMesos()
+
+    println "--> [mesos] creating new credentials... done"
+  }
+}
+
+def accountCreds = System.getenv("DCOS_SERVICE_ACCOUNT_CREDENTIAL")
```

Comment:
Should I assume the documentation for the format of this variable and it's JSON content are elsewhere?

Reply:
`DCOS_SERVICE_ACCOUNT_CREDENTIAL` contains the DC/OS service account info. It has the following fields:

```
{
    "scheme": "RS256",
    "uid": "jenkins-principal",
    "private_key": "<private-key-value>",
    "login_endpoint": "http://master.mesos/acs/api/v1/auth/login"
}
```

Reply:
I was asking mostly for future users of this code. How will they know what the file should look like? (Or is it a standard file and I'm just not remembering it today?)

Reply:
I copied the JSON above from the [Jenkins auth documentation](https://docs.mesosphere.com/services/jenkins/jenkins-auth/#using-the-web-interface-2). I am unable to find it anywhere. Suggestions on how to proceed? Should I add a link in a comment?

This script is not expected to be visible to the user.

Reply:
Does the user have to set the `DCOS_SERVICE_ACCOUNT_CREDENTIAL` value in the deployment? Or is that internal to DC/OS? If it's internal... maybe a comment that it's an internal variable. If it's set by the user... how will they know what value to use?

Reply:
The value is set by DCOS and the Universe config options trigger this being set. I will add a comment that it is internal and not provided by the user.

Reply:
+1 Well done finding an internal value to use. Makes setup much simpler!